File: benchmark/src/php/ack.php
<?php
function ack($m, $n) {
    if ($m == 0) return $n + 1;
    if ($n == 0) return ack($m - 1, 1);
    return ack($m - 1, ack($m, $n - 1));
}
$start = hrtime(true);
$result = ack(3, 6);
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Result: " . $result . "
";
echo "Time: " . round($ms) . "ms
";

File: benchmark/src/lua/ack.lua
local function ack(m, n)
    if m == 0 then
        return n + 1
    end
    if n == 0 then
        return ack(m - 1, 1)
    end
    return ack(m - 1, ack(m, n - 1))
end

local start = os.clock()
local result = ack(3, 6)
local finish = os.clock()
print("Result: " .. result)
print(string.format("Time: %.0fms", (finish - start) * 1000))

